packages/devextreme/js/__internal/ui/chat/messagebox.ts

```diff
@@ -36,20 +46,41 @@ class MessageBox extends DOMComponent<MessageBox, Properties> {
 
   _messageSendAction?: (e: Partial<MessageSendEvent>) => void;
 
+  _typingStartAction?: (e: Partial<TypingStartEvent>) => void;
+
+  _throttledTriggerTypingStartAction?: (e: Partial<TypingStartEvent>) => void;
+
+  _typingEndAction?: () => void;
+
+  _typingEndEventTimeout?: ReturnType<typeof setTimeout>;
+
   _getDefaultOptions(): Properties {
     return {
       ...super._getDefaultOptions(),
-      onMessageSend: undefined,
       activeStateEnabled: true,
       focusStateEnabled: true,
       hoverStateEnabled: true,
+      onMessageSend: undefined,
+      onTypingStart: undefined,
+      onTypingEnd: undefined,
     };
   }
 
   _init(): void {
     super._init();
 
     this._createMessageSendAction();
+    this._createTypingStartAction();
+    this._createTypingEndAction();
+
+    this._wrapTriggerTypingStartEvent();
+  }
+
+  _wrapTriggerTypingStartEvent(): void {
+    this._throttledTriggerTypingStartAction = throttle(
```

Comment:
This function is not an action.

Reply:
[30f954d8b2d174191c45c26c9ef35cce6e9da515]